statistics/dump.go

```diff
@@ -0,0 +1,189 @@
+}
+
+type jsonColumn struct {
+	Histogram         *tipb.Histogram `json:"histogram"`
+	CMSketch          *tipb.CMSketch  `json:"cm_sketch"`
+	AlreadyLoad       bool            `json:"already_load"`
+	NullCount         int64           `json:"null_count"`
+	LastUpdateVersion uint64          `json:"last_update_version"`
+}
+
+type jsonIndex struct {
+	Histogram         *tipb.Histogram `json:"histogram"`
+	CMSketch          *tipb.CMSketch  `json:"cm_sketch"`
+	NullCount         int64           `json:"null_count"`
+	LastUpdateVersion uint64          `json:"last_update_version"`
+}
+
+// ConvertTo converts Bucket's LowerBound and UpperBound to type t.
+func (b *Bucket) ConvertTo(h *Handle, t *types.FieldType) (err error) {
+	b.LowerBound, err = b.LowerBound.ConvertTo(h.ctx.GetSessionVars().StmtCtx, t)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	b.UpperBound, err = b.UpperBound.ConvertTo(h.ctx.GetSessionVars().StmtCtx, t)
+	return errors.Trace(err)
+}
+
+// DumpStatsToJSON dumps statistic to json.
+func (h *Handle) DumpStatsToJSON(dbName string, tableInfo *model.TableInfo) (*JSONTable, error) {
+	statsTable := h.statsCache.Load().(statsCache)[tableInfo.ID]
+	tbl, err := h.tableStatsFromStorage(tableInfo, true)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	jsonTbl := &JSONTable{
+		DatabaseName: dbName,
+		TableName:    tableInfo.Name.L,
+		Columns:      make(map[string]*jsonColumn, len(tbl.Columns)),
+		Indices:      make(map[string]*jsonIndex, len(tbl.Indices)),
+		Count:        tbl.Count,
+		ModifyCount:  tbl.ModifyCount,
+		Version:      tbl.Version,
+	}
+	for id, col := range tbl.Columns {
+		hist := &Histogram{
+			NDV:     col.NDV,
+			Buckets: make([]Bucket, len(col.Histogram.Buckets)),
+		}
+
+		// Convert Datum to BytesDatum.
+		for i, v := range col.Buckets {
+			hist.Buckets[i] = v
+			if err := hist.Buckets[i].ConvertTo(h, types.NewFieldType(mysql.TypeBlob)); err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+		// c is the Column in Cache.
+		c := statsTable.Columns[id]
+		jsonCol := &jsonColumn{
+			Histogram:         HistogramToProto(hist),
+			AlreadyLoad:       !(c.Count > 0 && len(c.Buckets) == 0),
+			NullCount:         col.NullCount,
+			LastUpdateVersion: col.LastUpdateVersion,
+		}
+		if col.CMSketch != nil {
+			jsonCol.CMSketch = CMSketchToProto(col.CMSketch)
+		}
+		jsonTbl.Columns[col.Info.Name.L] = jsonCol
+	}
+
+	for _, idx := range tbl.Indices {
+		jsonIdx := &jsonIndex{
+			Histogram:         HistogramToProto(&idx.Histogram),
+			NullCount:         idx.NullCount,
+			LastUpdateVersion: idx.LastUpdateVersion,
+		}
+		if idx.CMSketch != nil {
+			jsonIdx.CMSketch = CMSketchToProto(idx.CMSketch)
+		}
+		jsonTbl.Indices[idx.Info.Name.L] = jsonIdx
+	}
+	return jsonTbl, nil
+}
+
+func getCMSketch(c *tipb.CMSketch) (cms *CMSketch) {
+	if c != nil {
+		cms = CMSketchFromProto(c)
+		if c.Rows[0] == nil {
+			return cms
+		}
+		for _, val := range c.Rows[0].Counters {
+			cms.count += uint64(val)
```

Comment:
This is not needed because the `count` is already calculated in `CMSketchFromProto`.